File: scripts/reformat-hunpos.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-
# reformat hunpos tagged corpus output to nice list of 5grams
# also correct hunpos tags for single-character tokens like $
# the perl version of this was 4 times faster

import re, sys
hunpos_output = open("/home/tom/akb/cleaned-split-tagged",'r',1)
#hunpos_output = open("sample", 'r', 1)
newfile = open("entire_google_corpus", "w")
tagged_punct = re.compile(r'^(?P<single_char>[^ a-zA-Z0-9])\t(?P<tag>[^ ]+)\t')
wd_tag_pair = re.compile(r'^(\S+?)\s+(\S+?)\s+$')
sentence = ""
for word_tag in hunpos_output:   
   if (word_tag == "\n"):
      newfile.write(sentence + "\n")
      sentence = ""
   else:
      # avoid incorrectly tagged single-chars
      word_tag = tagged_punct.sub(r'\1\t\1\t', word_tag)
      # put in (word tag) form
      word_tag = wd_tag_pair.sub(r'(\1 \2) ', word_tag)
      sentence = sentence + word_tag   
newfile.close()
hunpos_output.close()

File: scripts/reformat_hunpos.pl
#!/usr/bin/perl
use warnings;
use strict;
# reformat hunpos tagged corpus output to nice list of 5grams
# also correct hunpos tags for single-character tokens like $

open (HUN_TAGS, '<', "/home/tom/akb/cleaned-split-tagged") or die "couldn't open input file: $!";
#open (HUN_TAGS, '<', "sample") or die "couldn't open input file: $!";
open (NEWONE, '>', 'entire_google') or die "couldn't open output file: $!";
my $sentence = "";
while (<HUN_TAGS>) {
    if (/^\n/) {
        $sentence = $sentence . "\n";
        print NEWONE $sentence;
        $sentence = "";
    } else {
        s/([^ a-zA-Z0-9])\t[^ ]+\t/($1 $1)/;
        s/([^ ]+?)\t([^ ]+?)\t/($1 $2)/;
        chomp;
        $sentence = $sentence . $_;
    }
    #print $sentence;        
}    
 close (HUN_TAGS) or die "can't close input file: $!";
 close (NEWONE) or die "can't close output file: $!";
